### PR TITLE
Dbex/73694 not sure

### LINKED ISSUE #73694: Toxic Exposure: Add "I'm not sure" to gulf war locations pages
## Issue Description
1. Add the "I'm not sure..." checkbox to the `gulf-war-1990` page and `gulf-war-2001`
- If "I'm not sure" and one or more locations is selected, nothing to do (it's as if not sure was not selected)
- Do not show a details page for "I'm not sure". 
- fyi, "I'm not sure" will be later ignored during back end submission
2. Add the "I'm not sure" checkbox to each of the details pages that display as a [view only field](https://depo-platform-documentation.scrollhelp.site/developer-docs/va-forms-library-how-to-include-view-only-data)
- same as above, will be ignored during back end processing

3. Add to summary page
- If a location does not have dates and has only "I'm not sure..." checkbox selected, display "I'm not sure of the dates"

In Scope:
- Gulf War 1990 pages
- Post 9/11 (Post 2001)

## Front end tasks
- Ensure the page is styled according to design spec. 
- ~Ensure the back end data from API is displayed in local and review instance~
- Ensure unit tests are available. 
- Ensure your code changes are covered by E2E tests
- Run axe checks using the Chrome or Firefox browser plugin
- Test for color contrast and color blindness issues
- Zoom layouts to 400% at 1280px width
- Test with 1 or 2 screen readers
- Navigate using the keyboard only

## Acceptance Criteria
- [ ] Designer comments on this issue to approve page styling 
- [ ] Unit tests pass
- [ ] E2E tests cover current code and regression
- [ ] End-to-end tests show 0 violations.
- [ ] ~The data returned matches API response (for given user or scenario)~
- [x] All axe checks pass
- [ ] All color contrast checks pass
- [x] All zoom testing passes
- [x] All keyboard checks pass 
- [x] All screenreader checks pass

## Appendix
<details> 

 <summary>Ensure your code changes are covered by E2E tests (expand)</summary>
 
 - Add E2E tests if none exist for this addition/change. 
 
 - Update existing E2E tests if this code will change functionality. 
 
 - Include axe checks, including hidden content
 
</details>

<details> 

 <summary> Run axe checks using the Chrome or Firefox browser plugin (expand)</summary>
 
- Ensure no heading levels are skipped.

- Ensure all buttons and labeled inputs use semantic HTML elements.

- Ensure all buttons, labeled elements and images are identified using HTML semantic markers or ARIA roles.

- Ensure form fields have clearly defined boundaries or outlines.

- Ensure form fields do not use placeholder text. 

- Ensure form fields have highly visible and specific error states. 

</details>

<details> 

 <summary> Test for color contrast and color blindness issues (expand) </summary>
 - All text has appropriate contrast. 

</details>

<details> 

 <summary> Zoom layouts to 400% at 1280px width (expand)</summary> 
 - Ensure readability and usability are supported when zoomed up to 400% at 1280px browser width
 - Ensure no content gets focused offscreen or is hidden from view.
 
</details>

<details> 

 <summary> Test with 1 or 2 screen readers (expand)</summary> 
 
 - Ensure the page includes a skip navigation link. 
 
 - Ensure all links are properly descriptive. 
 
 - Ensure screen reader users can hear the text equivalent for each image conveying information. 

 - Ensure screen reader users can hear the text equivalent for each image button. 

 - Ensure screen reader users can hear labels and instructions for inputs. 
 
 - Ensure purely decorative images are not announced by the screenreader. 
 
</details>

<details> 

 <summary>Navigate using the keyboard only (expand)</summary> 
 
 - Ensure all links (navigation, text and/or image), form controls and page functions can be reached with the tab key in a logical order. 
 
 - Ensure all links (navigation, text and/or image), form controls and page functions can be triggered with the spacebar, enter key, or arrow keys.
 
 - Ensure all interactive elements can be reached with the tab key in a logical order 
 
 - Ensure all interactive elements can be triggered with the spacebar, enter key, or arrow keys.
 
 - Ensure focus is always visible and appears in logical order.
 
 - Ensure each interactive element has visible focus state which appears in logical order.
 

</details>

---
## How to configure this issue
- [ ] **Attached to a Milestone** (when will this be completed?)
- [ ] **Attached to an Epic** (what body of work is this a part of?)
- [ ] **Labeled with Team** (`product support`, `analytics-insights`, `operations`, `service-design`, `Console-Services`, `tools-fe`)
- [ ] **Labeled with Practice Area** (`backend`, `frontend`, `devops`, `design`, `research`, `product`, `ia`, `qa`, `analytics`, `contact center`, `research`, `accessibility`, `content`)
- [ ] **Labeled with Type** (`bug`, `request`, `discovery`, `documentation`, etc.)

### PR DESCRIPTION
**Pull request by michael-wojcik**
## Summary

- Added `I'm not sure if I served in these locations` and `I'm not sure of the dates I served in this location` checkboxes to the `gulfWar1990` and `gulfWar2001` location and details pages, respectively
- Team: DEBX TREX
- Flipper: `disability_526_toxic_exposure`

## Related issue(s)

- department-of-veterans-affairs/va.gov-team#73694

## Testing done

- Location pages for `gulfWar1990` and `gulfWar2001` should display a new checkbox option `I'm not sure if I served in these locations`
- Details pages for `gulfWar1990` and `gulfWar2001` locations should display a new checkbox option `I'm not sure of the dates I served in this location`
- Summary pages for `gulfWar1990` and `gulfWar2001` location should display `I'm not sure of the dates` for locations where only `I'm not sure if I served in these locations` was selected on its respective details page

## Screenshots

| Before | After |
| ------ | ----- |
| Screenshot 2024-05-24 at 02 53 49 | Screenshot 2024-05-24 at 02 49 27 |
| Screenshot 2024-05-24 at 02 54 11 | Screenshot 2024-05-24 at 02 50 12 |
| | Screenshot 2024-05-24 at 02 50 38 |
| | Screenshot 2024-05-24 at 02 51 10 |

## What areas of the site does it impact?

526EZ

### Quality Assurance & Testing

- [x] I fixed|updated|added unit tests and integration tests for each feature (if applicable).
- [x] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs
- [x] Linting warnings have been addressed
- [x] Documentation has been updated ([link to documentation](#) \*if necessary)
- [x] Screenshot of the developed feature is added
- [x] [Accessibility testing](https://depo-platform-documentation.scrollhelp.site/developer-docs/wcag-2-1-success-criteria-and-foundational-testing) has been performed

### Error Handling

- [x] Browser console contains no warnings or errors.
- [x] Events are being sent to the appropriate logging solution

### Authentication

- [x] Did you login to a local build and verify all authenticated routes work as expected with a test user

### CODE REVIEW

**Review by mchae-nava — DISMISSED** [review]
LGTM, unit test coverage looks good at a glance!

File: src/applications/disability-benefits/all-claims/content/toxicExposureSummary.jsx [hunk 1]
```diff
@@ -28,20 +28,22 @@ export function toxicExposureSummary(
...
+                  <h5 className="vads-u-font-size--h6">
+                    {checkboxDefinitions[item]}
+                  </h5>
+                  <p className="vads-u-margin-y--1">
+                    {datesDescription(toxicExposure[datesObjectName][item])}
+                  </p>
+                </li>
+              )
+            );
+          })}
```
Comment (christinec-fftc) [inline src/applications/disability-benefits/all-claims/content/toxicExposureSummary.jsx:46]: This file was renamed and moved to the components dir. Could be part of the merge conflict.

File: src/applications/disability-benefits/all-claims/tests/pages/toxicExposure/gulfWar1990Locations.unit.spec.jsx [hunk 2]
```diff
@@ -109,4 +109,34 @@ describe('Gulf War 1990 Locations', () => {
...
+        uiSchema={uiSchema}
+        data={{}}
+        onSubmit={onSubmit}
+      />,
+    );
+
+    const checkboxGroup = $('va-checkbox-group', container);
+    await checkboxGroup.__events.vaChange({
+      target: { checked: true, dataset: { key: 'oman' } },
+      detail: { checked: true },
```
Comment (christinec-fftc) [inline src/applications/disability-benefits/all-claims/tests/pages/toxicExposure/gulfWar1990Locations.unit.spec.jsx:128]: there's some new test utils you can use here in `src/applications/mhv-secure-messaging/util/testUtils.js`

**Comment by christinec-fftc** [comment]
This feels like an odd scenario given that a location is not checked. It's an edge case, but wondering if we'd want to block the Veteran from continuing in this scenario? cc @Kate-allbee @mengA6 
image

File: src/applications/disability-benefits/all-claims/tests/pages/toxicExposure/gulfWar1990Locations.unit.spec.jsx (hunk 2 above)
Comment (michael-wojcik) [reply src/applications/disability-benefits/all-claims/tests/pages/toxicExposure/gulfWar1990Locations.unit.spec.jsx:128]: @christinec-fftc When trying to use the new `selectVaSelect` function from `src/applications/mhv-secure-messaging/util/testUtils.js`, this linter error is appearing:
```
No cross app imports allowed: disability-benefits importing from mhv-secure-messagingeslint(@department-of-veterans-affairs/no-cross-app-imports)
```

It seems like we can forgo using this for now, while that rule is in place. Do you agree?

**Comment by michael-wojcik** [comment]
> This feels like an odd scenario given that a location is not checked. It's an edge case, but wondering if we'd want to block the Veteran from continuing in this scenario? cc @Kate-allbee @mengA6 image

@christinec-fftc Should we move ahead with the current functionality so as not to block merging?

File: src/applications/disability-benefits/all-claims/tests/pages/toxicExposure/gulfWar1990Locations.unit.spec.jsx (hunk 2 above)
Comment (michael-wojcik) [reply src/applications/disability-benefits/all-claims/tests/pages/toxicExposure/gulfWar1990Locations.unit.spec.jsx]: I was able to use the new `checkVaCheckbox` method you added to `src/applications/disability-benefits/all-claims/tests/testUtils.js` instead

File: src/applications/disability-benefits/all-claims/constants.js [hunk 3]
```diff
@@ -374,6 +375,7 @@ export const GULF_WAR_2001_LOCATIONS = Object.freeze({
   yemen: 'Yemen',
   airspace: 'The airspace above any of these locations',
   none: 'None of these locations',
+  notsure: "I'm not sure if I served in these locations",
```
Comment (christinec-fftc) [inline src/applications/disability-benefits/all-claims/constants.js:378]:
```suggestion
  notsure: "I’m not sure if I served in these locations",
```

File: src/applications/disability-benefits/all-claims/content/toxicExposure.jsx [hunk 4]
```diff
@@ -67,6 +67,12 @@ export const startDateApproximate = 'Service start date (approximate)';
 export const endDateApproximate = 'Service end date (approximate)';
 export const goBackLink = 'Edit locations and dates';
 export const noDatesEntered = 'No dates entered';
+export const notSureDatesSummary = "I'm not sure of the dates";
```
Comment (christinec-fftc) [inline src/applications/disability-benefits/all-claims/content/toxicExposure.jsx:70]:
```suggestion
export const notSureDatesSummary = "I’m not sure of the dates";
```

File: src/applications/disability-benefits/all-claims/content/toxicExposure.jsx [hunk 5]
```diff
@@ -67,6 +67,12 @@ export const startDateApproximate = 'Service start date (approximate)';
...
+export const notSureDatesDetails = (
+  <p className="vads-spacing-1">
+    I'm not sure of the dates I served in this location
```
Comment (christinec-fftc) [inline src/applications/disability-benefits/all-claims/content/toxicExposure.jsx:73]:
```suggestion
    I’m not sure of the dates I served in this location
```

**Comment by christinec-fftc** [comment]
> > This feels like an odd scenario given that a location is not checked. It's an edge case, but wondering if we'd want to block the Veteran from continuing in this scenario? cc @Kate-allbee @mengA6 image
> 
> @christinec-fftc Should we move ahead with the current functionality so as not to block merging?

Let's check with design at 16th min today

**Comment by mengA6** [comment]
We discussed in 16th minute that we don't want to show an error in [this state](https://github.com/department-of-veterans-affairs/vets-website/pull/29975#issuecomment-2130436552). We should only show that error if they check "none of these locations" and an actual location.

**Comment by michael-wojcik** [comment]
> We discussed in 16th minute that we don't want to show an error in [this state](https://github.com/department-of-veterans-affairs/vets-website/pull/29975#issuecomment-2130436552). We should only show that error if they check "none of these locations" and an actual location.

Working to revise existing PR accordingly

**Review by christinec-fftc — APPROVED** [review]

**Pull request merged**